crypto/x509/x509_v3.c

```diff
@@ -118,6 +118,20 @@ STACK_OF(X509_EXTENSION) *X509v3_add_ext(STACK_OF(X509_EXTENSION) **x,
     } else
         sk = *x;
 
+    /*
+     * Empty OCTET STRINGs and empty SEQUENCEs encode to just two bytes of tag
+     * (0x04 or 0x30) and length (0x00).
+     */
+    if (ex->value.length == 2
+        && (ex->value.data[0] == 0x30 || ex->value.data[0] == 0x04)) {
```

Comment:
Also here, the part "&& (ex->value.data[0] == 0x30 || ex->value.data[0] == 0x04" could be dropped
because it will be implied by a positive outcome of the following `OBJ_cmp(obj, skid) == 0 || OBJ_cmp(obj, akid) == 0`